### ISSUE TITLE
Replace Mockito extension's with service loader API

### ISSUE THREAD

**Issue by raphw** [mockito/mockito#652]
Mockito is demanind a Java version 6+ meaning that we can now rely on the service loader API which was added with this release. This allows us to retire the Mockito-specific extension and use the service loader API instead.

This will also help us with Java 9 where Java module's require the service loader to be used for cross-module communication.

**Comment by TimvdLippe** [comment]
Will this be a breaking change or can we make it backwards-compatible?

**Comment by raphw** [comment]
Depends. We could offer loading from both locations but I assume we would like to substitute the mechanism. I wonder if we only include this in 3, there are several details to work out.

**Comment by bric3** [comment]
Nope, the custom service-loader was made because Android guys don't have a working service loader. When their app is packaged the META-INF folder was removed. I don't know at this moment. But chances it will break older Android development.

**Comment by bric3** [comment]
For reference : https://code.google.com/p/android/issues/detail?id=59658
